test/Test.hs

```diff
@@ -22,8 +31,405 @@ monadReader2 = assertEqual "should be equal" [(5, 0)] $
       y <- ask
       return (x,y)
 
+monadReader3 :: Assertion
+monadReader3 = assertEqual "should be equal" [5,3] $
+  runReader (observeAllT (plus5 `mplus` mzero `mplus` plus3)) (0 :: Int)
+  where
+    plus5 = local (5+) ask
+    plus3 = local (3+) ask
+
+nats, odds, oddsOrTwo,
+  oddsOrTwoUnfair, oddsOrTwoFair,
+  odds5down :: Monad m => LogicT m Integer
+
+nats = pure 0 `mplus` ((1 +) <$> nats)
+
+odds = return 1 `mplus` liftM (2+) odds
+
+oddsOrTwoUnfair = odds `mplus` return 2
+oddsOrTwoFair   = odds `interleave` return 2
+
+oddsOrTwo = do x <- oddsOrTwoFair
+               if even x then once (return x) else mzero
+
+odds5down = return 5 `mplus` mempty `mplus` mempty `mplus` return 3 `mplus` return 1
+
+yieldWords :: [String] -> LogicT m String
+yieldWords = go
+  where go [] = mzero
+        go (w:ws) = return w `mplus` go ws
+
+
 main :: IO ()
-main = defaultMain $ testGroup "All"
+main = defaultMain $
+  localOption (mkTimeout 1000000) $  -- 1 second
```

Comment:
I think it's better to apply this option to specific, `expectFail` tests only.

Reply:
I can move the timeout to those tests, but my thought for applying it globally was that if someone (or something) made a change that caused divergence, this would cause the test suite to fail rather than run forever which would necessitate human interaction or CI timeouts to deduce failures.  Let me know if that sounds reasonable or if you'd still like this moved to just the tests that need them.

Reply:
OK, let's keep it as is.